src/databricks/model_evaluation.py

```diff
@@ -0,0 +1,695 @@
+dfs = [
+    heating_metrics_by_type_upgrade,
+    cooling_metrics_by_type_upgrade,
+    total_metrics_by_upgrade,
+    total_metrics_by_fuel,
+]
+metrics_by_upgrade_type_model = reduce(DataFrame.unionByName, dfs)
+
+# COMMAND ----------
+
+# DBTITLE 1,Save metrics table
+if not DEBUG:
+    # save the metrics table tagged with the model name and version number
+    (
+        metrics_by_upgrade_type_model.write.format("delta")
+        .mode("overwrite")
+        .option("overwriteSchema", "true")
+        .option("userMetadata", MODEL_RUN_NAME)
+        .saveAsTable("ml.surrogate_model.evaluation_metrics")
+    )
+
+# COMMAND ----------
+
+# MAGIC %md ## Format for export
+
+# COMMAND ----------
+
+# DBTITLE 1,Convert to pandas
+metrics_by_upgrade_type_model_pd = metrics_by_upgrade_type_model.toPandas()
+
+# COMMAND ----------
+
+# DBTITLE 1,Rename columns for readabiity
+# rename metric columns to be more readable
+metric_rename_dict = {
+    "median_absolute_percentage_error_savings": "Median APE - Savings",
+    "mean_absolute_percentage_error_savings": "Mean APE - Savings",
+    "median_absolute_error_savings": "Median Abs Error - Savings",
+    "mean_absolute_error_savings": "Mean Abs Error - Savings",
+    "weighted_mean_absolute_percentage_error_savings": "Weighted Mean APE - Savings",
+    "median_absolute_percentage_error": "Median APE",
+    "mean_absolute_percentage_error": "Mean APE",
+    "median_absolute_error": "Median Abs Error",
+    "mean_absolute_error": "Mean Abs Error",
+    "weighted_mean_absolute_percentage_error": "Weighted Mean APE",
+}
+
+key_rename_dict = {"upgrade_id": "Upgrade ID", "type": "Type", "model": "Model"}
+
+metrics_by_upgrade_type_model_pd.rename(
+    columns={**metric_rename_dict, **key_rename_dict}, inplace=True
+)
+
+# COMMAND ----------
+
+# DBTITLE 1,Specify row order
+# set the order in which types will appear in the table
+metrics_by_upgrade_type_model_pd["Type"] = pd.Categorical(
+    metrics_by_upgrade_type_model_pd["Type"],
+    categories=[
+        "Electricity",
+        "Electric Resistance",
+        "Methane Gas",
+        "Propane",
+        "Fuel Oil",
+        "Shared Heating",
+        "No Heating",
+        "None",
+        "Heat Pump",
+        "AC",
+        "Room AC",
+        "Shared Cooling",
+        "No Cooling",
+        "Total",
+    ],
+    ordered=True,
+)
+
+# COMMAND ----------
+
+# DBTITLE 1,Pivot to wide on model type and sort by upgrade and type
+metrics_by_upgrade_type_model_metric = metrics_by_upgrade_type_model_pd.melt(
+    id_vars=["Type", "Model", "Upgrade ID"],
+    value_vars=list(metric_rename_dict.values()),
+    var_name="Metric",
+)
+
+metrics_by_upgrade_type = metrics_by_upgrade_type_model_metric.pivot(
+    index=[
+        "Upgrade ID",
+        "Type",
+    ],
+    columns=["Metric", "Model"],
+    values="value",
+).sort_values(["Upgrade ID", "Type"])
+
+# COMMAND ----------
+
+# DBTITLE 1,Display Results
+metrics_by_upgrade_type
```

Comment:
My sense was that this was mostly just due to the fact that the savings are larger for 4 than 3, and so when using these relative measures, 4 looks better than 3. If you look at the error for just the consumption rather than savings  [here](https://docs.google.com/spreadsheets/d/1XO9a_NPOcqaZrsisJSCs5T3R2enVSPJ-eKI1Ilgf-H8/edit?gid=86053945#gid=86053945) (the right half), you'll see that they are quite similar. But there may be something deeper going on and I just don't realize it..

Reply:
Ahh that's a good call - I didn't think of it from that point of view